cmake/igl/igl_install.cmake

```diff
@@ -61,3 +63,10 @@ function(igl_install module_name)
         )
     endforeach()
 endfunction()
+
+# These two variables keep track whether a restricted or copyleft target
+# was installed, so that the corresponding install/export call can be made
+# at the time of project installation.
+# See top level CMakeLists.txt in root folder of the libigl repo.
+set(LIBIGL_INSTALLED_RESTRICTED  FALSE CACHE INTERNAL "Tracks whether a restricted target was installed")
```

Comment:
I don't think these will work the way you intend. Maybe we should create an empty export set `LibiglTargets${suffix}` for each possible suffix, and simply unconditionally export that at the end of the root CMakeLists? That way if the set is empty, nothing will be exported anyway?

Reply:
In my tests, it worked as expected, but I could very well be blind to a side effect.  I tried something similar to what you suggest, but it seems CMake will throw if you try to call install/export with an export list that does not exist.  I didn't know how to create an empty export list, but that seems like a better solution if possible.  I will do some searching to figure that out.

Reply:
The problem I see with using a CACHE variable is that once `LIBIGL_INSTALLED_COPYLEFT` is set to `FALSE`, subsequent calls to `set(LIBIGL_INSTALLED_COPYLEFT ...)` will not change the value to `TRUE`. When you check your `CMakeCache.txt` after running CMake I suspect the value of this variable should still be `FALSE`.

Reply:
I did check my CMakeCache and it does show them as TRUE (when I have a copyleft or restricted option turned on respectively). It appears to be working as I expect with v3.25.1. However, I have not tested it with older versions.

Reply:
But...I agree that an empty export set would be cleaner.

Reply:
Is this PR still in progress?